lib/utils/task_queue/worker_queue.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:fluffychat/utils/task_queue/task.dart';
+import 'package:matrix/matrix.dart';
+
+typedef OnTaskCompleted = void Function(String? taskId);
+
+abstract class WorkerQueue {
+  final Queue<Task> _queue = Queue<Task>();
+  Completer? _completer;
+
+  String get workerName;
+
+  Queue<Task> get queue => _queue;
+
+  Task? _runningTask;
+
+  Future addTask(Task task) {
+    _queue.add(task);
+    Logs().i(
+      'WorkerQueue<$workerName>::addTask(): QUEUE_LENGTH: ${_queue.length}',
+    );
+    return _processTask();
+  }
+
+  bool isTaskIdInQueue(String taskId) {
+    return _queue.any((task) => task.id == taskId) ||
+        _runningTask?.id == taskId;
+  }
```

Comment:
I can not find any place to call it? Can we remove it?

Reply:
removed